// This file defines how to split FHIR resources into versioned resources.

syntax = "proto3";

package google.fhir.proto;

option go_package = "github.com/google/fhir/go/proto";

message ResourceConfig {
  // If not specified otherwise, all events in this resource are assumed to
  // have occurred at the timestamp in this field.
  repeated string default_timestamp_fields = 1;

  // TODO: Add mechanism for specifying when contained resources
  // should be added in to a versioned resource.

  message TimestampOverride {
    string timestamp_field = 1;
    repeated string resource_field = 2;
  }
  repeated TimestampOverride timestamp_override = 5;

  reserved 2, 3, 4;
}

message VersionConfig {
  map<string, ResourceConfig> resource_config = 1;
}
